### PR TITLE
haproxy: Fix VIP creation for haproxy

### PR DESCRIPTION
**Pull request by rhafer**
Up to now the haproxy resource couldn't cleanly cope with with new
virtual IPs being added for haproxy after the initial deployment.
Since we are using "pacemaker_transaction" for creating new resources
and "pacemaker_primitive" for updating existing ones there was an
ordering issue when a new VIP was added when the haproxy group already
existed. The :update for the group was not executed since VIP was not
yet created.

With MariaDB support now available this is something that can happen
quite regulary. MariaDB only needs a VIP in the "admin" network, if
later "Keystone" is deployed to the same cluster it needs an additional
VIP for the "public" network.

We're trying to fix this shortcomming now by creating the VIPs outside
of the transaction (but with target-role="Stopped") and explicitly
starting them after the transaction ran.

Related-Bug: bsc#1056371

### CODE REVIEW

**Comment by rhafer** [comment]
With this the VIP is again started before the group is updated, which causes pacemaker to shuffle around resources, when the group is updated :/

**Comment by rhafer** [comment]
Adding the do not merge label to add a wait loop which checks for the VIPs/haproxy being started.

**Comment by rhafer** [comment]
> Adding the do not merge label to add a wait loop which checks for the VIPs/haproxy being started.

Adding a wait loop is hardly possible and very racey. But https://github.com/crowbar/crowbar-ha/pull/234 does the trick in much simpler way.

File: chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb [hunk 1]
```diff
@@ -56,6 +56,19 @@
   cluster_vhostname = CrowbarPacemakerHelper.cluster_vhostname(node)
   service_name = "haproxy"
 
+  # Make sure new pacemaker resources get created before before
+  # the group ("g-haproxy") is updated. Otherwise we run into ordering
+  # issue when sub-sequent chef-client runs need to create additional
```
Comment (s-t-e-v-e-n-k) [inline chef/cookbooks/crowbar-pacemaker/recipes/haproxy.rb]: Nit: subsequent is not a word that requires hyphenation.

**Pull request merged**